### PR TITLE
DMVP-1145: Refactored Ingress module

### PR DESCRIPTION
**Pull request by viktoryathegreat**
This module has to create a k8s ingress resource which will create an alb in AWS but somehow the current setup doesn’t support this functionality anymore.

Refactored the module.

### CODE REVIEW

File: modules/ingress/variables.tf [hunk 1]
```diff
@@ -15,15 +15,17 @@ variable "annotations" {
...
+    service_port_number = string
+    service_port_name   = string
+    path                = string
   }))
   default = [
     {
-      service_name = "dummy"
-      service_port = "80"
-      path         = "/alb-terraform-created"
+      service_name        = "response-200"
```
Comment (aramkarapetian) [inline modules/ingress/variables.tf:25]: I suggest we hardcode this in the module as user does not need to know about limitations and should only supply own variables.

File: modules/ingress/variables.tf [hunk 2]
```diff
@@ -33,8 +35,8 @@ variable "default_backend" {
     service_port = string
   })
   default = {
-    service_name = "dummy"
-    service_port = "80"
+    service_name = null
```
Comment (aramkarapetian) [inline modules/ingress/variables.tf:38]: I think we don't need to have default values or those should not be null as not supplying values should not result in strange behaviour.
If those values are mandatory then we could make variables out of those and make them mandatory there.

File: modules/ingress/README.md [hunk 3]
```diff
@@ -40,20 +43,28 @@ module "ingress" {
...
   annotations = {
-       "alb.ingress.kubernetes.io/load-balancer-name" = "test-ingress"
-       "alb.ingress.kubernetes.io/scheme"             = "internet-facing"
-       "alb.ingress.kubernetes.io/subnets"            = "subnet-0ebc13842a5f, subnet-05b72a3769b, subnet-0da2ece4bb4229"
-       "alb.ingress.kubernetes.io/backend-protocol"   = "HTTP"
-       "alb.ingress.kubernetes.io/certificate-arn"    = "arn:aws:acm:us-east-1:5********68:certificate/a55ee6eb****1706"
-       "alb.ingress.kubernetes.io/group.name"         = "dev"
-       "alb.ingress.kubernetes.io/listen-ports"       = "[{\"HTTPS\":443}, {\"HTTPS\":80}]"
-       "kubernetes.io/ingress.class"                  = "alb"
+       "alb.ingress.kubernetes.io/load-balancer-name"   = "test-ingress"
```
Comment (aramkarapetian) [inline modules/ingress/README.md:46]: I suggest we move those into separate variables and merge inside the module.
When using the module I don't need to understand how this is constructed and also if we refactor the module and use some other mechanism that will force module signature update when ingress module should be implementation agnostic.

File: modules/ingress/README.md [hunk 4]
```diff
@@ -1,61 +1,39 @@
 # Module setup ingress controller.
 
-# Example 1. Minimal parameter set and create ingress on default annotations
+## Problem
+When creating R53 records or CDN origins we need to explicitly pass ALB's DNS name to those resources. It's not convenient and not flexible.
```
Comment (aramkarapetian) [inline modules/ingress/README.md:4]: Please ask someone to give feedback on the text, I do know the concept and can understand but not sure if everyone will understand.

File: modules/ingress/README.md [hunk 5]
```diff
@@ -1,61 +1,39 @@
...
-  api_version = extensions/v1beta1
+module "ingress" {
+  source = "dasmeta/modules/aws//modules/ingress"
 
-  annotations = {
-    "alb.ingress.kubernetes.io/certificate-arn"    = "arn:aws:acm:us-east-1:5********68:certificate/a55ee6eb****1706"
-  }
-}
-```
+  name     = "terraform-ingress
```
Comment (aramkarapetian) [inline modules/ingress/README.md:29]: Missing double quote here.

File: modules/ingress/README.md [hunk 6]
```diff
@@ -1,61 +1,39 @@
...
+1. creates a k8s Ingress resource,
+2. created resource makes an ALB in AWS,
+3. has `alb.ingress.kubernetes.io/group.name` annotation set by default.
 
-  annotations = {
-    "alb.ingress.kubernetes.io/certificate-arn"    = "arn:aws:acm:us-east-1:5********68:certificate/a55ee6eb****1706"
-  }
-}
-```
+In the result of this you can
```
Comment (aramkarapetian) [inline modules/ingress/README.md:13]: You probably want ":" ate the end of the sentence.

File: modules/ingress/README.md [hunk 7]
```diff
@@ -1,61 +1,39 @@
...
-       "kubernetes.io/ingress.class"                  = "alb"
-     }
-  path = [
-    {
-      service_name = "nginx"
-      service_port = "80"
-      path         = "/alb-terraform-created"
-    }
-  ]
+  tls_hosts = "dasmeta.com"
```
Comment (aramkarapetian) [inline modules/ingress/README.md:36]: Don't we want to give working example to the user? I mean ingress manifest that can link with the main ingress by group. We do have it in examples folder but better have it here to have it complete.

**Review by mrdntgrn — APPROVED** [review]

**Pull request merged**